### PR TITLE
Removed res channel and added lock for checkpoint update

### PR DESCRIPTION
**Pull request by jeongkyun-oh**
## Proposed changes

- Removed response loop to simplify logic and remove dependency between loops

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: datasync/chaindatafetcher/types.go [hunk 1]
```diff
@@ -38,32 +34,19 @@ const (
 
 // request contains a raw block which should be handled and the type of data which should be exported.
```
Comment (ehnuje) [inline datasync/chaindatafetcher/types.go:35]:
```suggestion
// request contains a blockNumber which should be handled and the type of data which should be exported.
```

File: datasync/chaindatafetcher/chaindata_fetcher.go [hunk 2]
```diff
@@ -134,12 +130,12 @@ func (f *ChainDataFetcher) Stop() error {
 	return nil
 }
 
-func (f *ChainDataFetcher) makeRequests(start, end uint64, reqType requestType, stopCh chan struct{}) {
+func (f *ChainDataFetcher) makeRequests(start, end uint64, reqType requestType, updateCheckpoint bool, stopCh chan struct{}) {
```
Comment (ehnuje) [inline datasync/chaindatafetcher/chaindata_fetcher.go:133]: Using both `newRequest` and `makeRequests` confuses me. 
How about changing `makeRequests` to `sendRequests`?
```suggestion
func (f *ChainDataFetcher) sendRequests(start, end uint64, reqType requestType, updateCheckpoint bool, stopCh chan struct{}) {
```

File: datasync/chaindatafetcher/chaindata_fetcher.go [hunk 3]
```diff
@@ -251,7 +247,7 @@ func (f *ChainDataFetcher) SetComponents(components []interface{}) {
 	}
 }
 
-func (f *ChainDataFetcher) handleRequestByType(reqType requestType, ev blockchain.ChainEvent) {
+func (f *ChainDataFetcher) handleRequestByType(reqType requestType, updateCheckpoint bool, ev blockchain.ChainEvent) {
```
Comment (ehnuje) [inline datasync/chaindatafetcher/chaindata_fetcher.go:250]: boolean value `updateCheckpoint` is not used.

Comment (ehnuje) [inline datasync/chaindatafetcher/chaindata_fetcher.go:250]:
```suggestion
func (f *ChainDataFetcher) handleRequestByType(reqType requestType, shouldUpdateCheckpoint bool, ev blockchain.ChainEvent) {
```

File: datasync/chaindatafetcher/chaindata_fetcher.go [hunk 4]
```diff
@@ -134,12 +130,12 @@ func (f *ChainDataFetcher) Stop() error {
 	return nil
 }
 
-func (f *ChainDataFetcher) makeRequests(start, end uint64, reqType requestType, stopCh chan struct{}) {
+func (f *ChainDataFetcher) sendRequests(start, end uint64, reqType requestType, shouldUpdateCheckpoint bool, stopCh chan struct{}) {
 	for i := start; i <= end; i++ {
 		select {
 		case <-stopCh:
 			return
```
Comment (ehnuje) [inline datasync/chaindatafetcher/chaindata_fetcher.go:137]: It would be better to log to indicate that the sendRequests successfully receives `stopCh`

Comment (ehnuje) [reply datasync/chaindatafetcher/chaindata_fetcher.go:137]: Not necessarily need to be done in this PR.

Comment (jeongkyun-oh) [reply datasync/chaindatafetcher/chaindata_fetcher.go:137]: Right. I will add some logs on the next PR.

**Comment by jeongkyun-oh** [comment]
@aidan-kwon @KimKyungup PTAL.

**Pull request merged**